tests/python/unittest/test_gluon_rnn.py

```diff
@@ -152,190 +124,7 @@ def test_lstmp():
-    inputs = [mx.sym.Variable('t%d_data'%i) for i in range(3)]
-    outputs, _ = cell.unroll(3, inputs)
-    outputs = mx.sym.Group(outputs)
-    keys = sorted(cell.collect_params().keys())
-    for i in range(5):
-        if i==1:
-            continue
-        assert '%d.h2h_weight'%i in keys
-        assert '%d.h2h_bias'%i in keys
-        assert '%d.i2h_weight'%i in keys
-        assert '%d.i2h_bias'%i in keys
-    assert '1.base_cell.h2h_weight' in keys
-    assert '1.base_cell.h2h_bias' in keys
-    assert '1.base_cell.i2h_weight' in keys
-    assert '1.base_cell.i2h_bias' in keys
-    assert outputs.list_outputs() == ['lstmcell_t0_out_output', 'lstmcell_t1_out_output', 'lstmcell_t2_out_output']
-
-    args, outs, auxs = outputs.infer_shape(t0_data=(10,50), t1_data=(10,50), t2_data=(10,50))
-    assert outs == [(10, 100), (10, 100), (10, 100)]
-
-
-@pytest.mark.serial
-def test_hybridstack():
-    cell = gluon.rnn.HybridSequentialRNNCell()
-    for i in range(5):
-        if i == 1:
-            cell.add(gluon.rnn.ResidualCell(gluon.rnn.LSTMCell(100)))
-        else:
-            cell.add(gluon.rnn.LSTMCell(100))
-    inputs = [mx.sym.Variable('t%d_data'%i) for i in range(3)]
-    outputs, _ = cell.unroll(3, inputs)
-    outputs = mx.sym.Group(outputs)
-    keys = sorted(cell.collect_params().keys())
-    for i in range(5):
-        if i==1:
-            continue
-        assert '%d.h2h_weight'%i in keys
-        assert '%d.h2h_bias'%i in keys
-        assert '%d.i2h_weight'%i in keys
-        assert '%d.i2h_bias'%i in keys
-    assert '1.base_cell.h2h_weight' in keys
-    assert '1.base_cell.h2h_bias' in keys
-    assert '1.base_cell.i2h_weight' in keys
-    assert '1.base_cell.i2h_bias' in keys
-    assert outputs.list_outputs() == ['lstmcell_t0_out_output', 'lstmcell_t1_out_output', 'lstmcell_t2_out_output']
-
-    args, outs, auxs = outputs.infer_shape(t0_data=(10,50), t1_data=(10,50), t2_data=(10,50))
-    assert outs == [(10, 100), (10, 100), (10, 100)]
-
-    # Test HybridSequentialRNNCell nested in nn.HybridBlock, SequentialRNNCell will fail in this case
-    class BidirectionalOfSequential(gluon.HybridBlock):
-        def __init__(self):
-            super(BidirectionalOfSequential, self).__init__()
-
-            cell0 = gluon.rnn.HybridSequentialRNNCell()
-            cell0.add(gluon.rnn.LSTMCell(100))
-            cell0.add(gluon.rnn.LSTMCell(100))
-
-            cell1 = gluon.rnn.HybridSequentialRNNCell()
-            cell1.add(gluon.rnn.LSTMCell(100))
-            cell1.add(gluon.rnn.LSTMCell(100))
-
-            self.rnncell = gluon.rnn.BidirectionalCell(cell0, cell1)
-
-        def hybrid_forward(self, F, x):
-            return self.rnncell.unroll(3, x, layout="NTC", merge_outputs=True)
-
-    x = mx.nd.random.uniform(shape=(10, 3, 100))
-    net = BidirectionalOfSequential()
-    net.initialize()
-    outs, _ = net(x)
-
-    assert outs.shape == (10, 3, 200)
-
-
-def test_bidirectional():
```

Comment:
Should this test be adapted to testing the front-end infer_shape functions instead?